### ISSUE TITLE
RemoveMarkers doesnt work

### ISSUE THREAD

**Issue by matthijssn** [Yermo/nativescript-mapbox#229]
In my MapReady event (Angular Typescript project) I try to remove all markers before adding a new collection. This doesnt seem to work, the new markers are put on top of the old ones.

//1) Remove old ones
  map.removeMarkers();                
//1) Add new marker collection
  map.addMarkers(markers);

**Comment by EddyVerbruggen** [comment]
Can you share a repo reproducing this issue?

**Comment by matthijssn** [comment]
```
  onMapReady(args) {      
        var map : MapboxViewApi = args.map;
        console.log("Map ready!"); 
        const nativeMapView = args.ios ? args.ios : args.android;
        console.log(`Mapbox onMapReady for ${args.ios ? "iOS" : "Android"}, native object received: ${nativeMapView}`);             
        this._socketIO = SocketIO.connect(Config.server_url, <SocketIO.SocketOptions>{
            log: true 
        }); 
       
        this._socketIO.emit("updatecoordinates", 
                                        {
                                            'minLat': 51.472874, 
                                            'minLong' : 3.520436, 
                                            'maxLat' : 53.686143, 
                                            'maxLong' : 7.090418 
                                        });


        this._socketIO.on("locationdata", function(data) {    
              
                var markers = []                         
                data.forEach(plane => {  
                    markers.push(    
                        {
                            lat: plane.latitude,
                            lng: plane.longitude,                        
                            title: plane.callsign,
                            subtitle: plane.origin_country,
                            selected: false, 
                            onCalloutTap: function(){console.log("'Test");}
                        });
                });    
                      
                map.removeMarkers();
                map.addMarkers(markers); 
            }
        );  
        
      }      
```

in the map.removeMarkers() nothing seems to happen.

**Comment by EddyVerbruggen** [comment]
That's not a repo..

And `removeMarkers` returns a Promise. Does it work better if you wait for it to resolve before adding markers?

Also: which platform(s)?

**Comment by matthijssn** [comment]
With a repo you mean that I share the complete project so you can run it? I thought that just the mapReady event should be sufficient. I'll try the Promise way.

**Comment by EddyVerbruggen** [comment]
I mean that if you want me to make sense of an issue like this it's very convenient to be able to clone a repo that reproduces it. I don't want your entire project of course.

Btw, the demo app has an example of the remove all feature. Did you try cloning and running it?

**Comment by matthijssn** [comment]
Working now on the Promise structure as can be found in the demo. Thanks! I'll post the result.

**Comment by EddyVerbruggen** [comment]
Cool 👍

Btw I'm not saying there isn't an issue but with 10 million being reported in my repos every day I really need an easy way to recreate them in order to be able to find time to fix any of them.

**Comment by EddyVerbruggen** [comment]
If the issue was with `removeAllMarkers` on Android, then this is fixed in 4.1.0.
